examples/transformers_text.py

```diff
@@ -92,18 +107,51 @@
 
 class TextToEmbedding:
     def __init__(self, model: str, pooling: str, device: torch.device):
-        self.tokenizer = AutoTokenizer.from_pretrained(model)
-        self.model = AutoModel.from_pretrained(model).to(device)
+        self.model_name = model
         self.device = device
-        self.pooling = pooling
+        self.tokenizer = AutoTokenizer.from_pretrained(model)
+        if model == "intfloat/e5-mistral-7b-instruct":
+            self.pooling = "last"
```

Comment:
It's a bit weird not to respect the `pooling` arg here.

Reply:
For decoder model (causal LLM), only using the last token embedding (as the sentence embedding) makes sense, so I force using "last" pooling here.

Reply:
makes sense.